packages/nextjs/utils/scaffold-eth/networks.ts

```diff
@@ -54,40 +55,46 @@ export const NETWORKS_EXTRA_DATA: Record<string, TChainAttributes> = {
  * @param txnHash
  * @dev returns empty string if the network is localChain
  */
-export const getBlockExplorerTxLink = (network: Network, txnHash: string) => {
-  const { name, chainId } = network;
+export function getBlockExplorerTxLink(network: Network, txnHash: string) {
+  let blockExplorerTxURL = "";
+  const { chainId } = network;
 
-  if (chainId === 31337 || chainId === 1337) {
-    // If its localChain then return empty sting
+  const chainNames = Object.keys(chains);
+
+  const targetChainArr = chainNames.filter(chainName => {
+    const wagmiChain = chains[chainName as keyof typeof chains];
+    return wagmiChain.id === chainId;
+  });
+
+  if (targetChainArr.length === 0) {
     return "";
   }
 
-  let blockExplorerNetwork = "";
-  if (name && chainId > 1) {
-    blockExplorerNetwork = name + ".";
-  }
+  const targetChain = targetChainArr[0] as keyof typeof chains;
+  // @ts-expect-error : ignoring error since `blockExplorers` key may or may not be present and we are already checking in ternary
+  blockExplorerTxURL = chains[targetChain]?.blockExplorers?.default?.url
+    ? // @ts-expect-error
+      chains[targetChain].blockExplorers.default.url
+    : "";
 
-  let blockExplorerBaseTxUrl = "https://" + blockExplorerNetwork + "etherscan.io/tx/";
-  if (chainId === 100) {
-    blockExplorerBaseTxUrl = "https://blockscout.com/poa/xdai/tx/";
+  if (!blockExplorerTxURL) {
+    return "";
   }
 
-  const blockExplorerTxURL = blockExplorerBaseTxUrl + txnHash;
-
-  return blockExplorerTxURL;
-};
+  return `${blockExplorerTxURL}/tx/${txnHash}`;
+}
 
 /**
  * Get the wagmi's Chain target network configured in the app.
  */
 export const getTargetNetwork = () => {
```

Comment:
After thinking about it a bit, completely removed this function at [eb6d127](https://github.com/scaffold-eth/se-2/pull/246/commits/eb6d127ba97a6554eccecf3b6cf663d8b39fbf66)

Reason : I think it might create ambiguity and confusion weather `getTargetNetwork` should be used or `scaffolConfig.targeNetwork` should be used. 

Also this is how it was defined which I think don't add value now since we have great chain auto completion in `scaffold.config.ts`

```ts
export const getTargetNetwork = () => {
  const network = scaffoldConfig.targetNetwork;

  if (!network) {
    // If error defaults to hardhat local network
    console.error("Network name is not set, misspelled or unsupported network used in .env.*");
    return chains.hardhat;
  }

  return network;
};
```

Reply:
Simpler is probably better, less risk of causing confusion. :+1:

Reply:
Yes, I agree. Removing it was a good decision!

One thing I'm not sure about: inlining `scaffoldConfig.targetNetwork` or `scaffoldConfig.targetNetwork.id` or `scaffoldConfig.targetNetwork.name` directly. In some cases I think I'd have done something like:

```ts
const configuredNetwork = scaffoldConfig.targetNetwork;

// use configuredNetwork
```

Specially on places like `ContractUI`, where it's used several times. But I guess it's redundant if we are only using it once, right? 

Anyway, a nitpick, you can ignore me :D